homeassistant/components/vallox/number.py

```diff
@@ -0,0 +1,159 @@
+"""Support for Vallox ventilation unit numbers."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from vallox_websocket_api import Vallox
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ValloxDataUpdateCoordinator, ValloxEntity
+from .const import DOMAIN
+
+
+class ValloxNumberEntity(ValloxEntity, NumberEntity):
+    """Representation of a Vallox number entity."""
+
+    entity_description: ValloxNumberEntityDescription
+    _attr_has_entity_name = True
+    _attr_entity_category: EntityCategory | None = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxNumberEntityDescription,
+        client: Vallox,
+    ) -> None:
+        """Initialize the Vallox number entity."""
+        super().__init__(name, coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self._device_uuid}-{description.key}"
+        self._client = client
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the value reported by the sensor."""
+        if (
+            value := self.coordinator.data.get_metric(
+                self.entity_description.metric_key
+            )
+        ) is None:
+            return None
+
+        return float(value)
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Update the current value."""
+        await self._client.set_values(
+            {self.entity_description.metric_key: float(value)}
+        )
+        await self.coordinator.async_request_refresh()
+
+
+@dataclass
+class ValloxNumberEntityDescription(NumberEntityDescription):
+    """Describes Vallox number entity."""
+
+    metric_key: str = ""
+    entity_type: type[ValloxNumberEntity] = ValloxNumberEntity
```

Comment:
We are not subclassing `ValloxNumberEntity` (yet), so we can remove this field for now and instantiate everything as `ValloxNumberEntity` unconditionally.